lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -0,0 +1,176 @@
+                for child_key in xblock.children:
+                    summary = summarize_block(child_key)
+                    block_structure.set_transformer_block_data(child_key, cls, 'block_analytics_summary', summary)
+
+    def transform(self, user_info, block_structure):
+        """
+        Mutates block_structure and block_data based on the given user_info.
+
+        Arguments:
+            user_info(object)
+            block_structure (BlockStructureCollectedData)
+        """
+
+        def publish_events(location, previous_count, max_count, block_keys):
+            """
+            Helper method to publish events for analytics purposes
+            """
+
+            def format_block_keys(keys):
+                """ Helper method to format block keys """
+                json = []
+                for key in keys:
+                    info = block_structure.get_transformer_block_data(block_key, self, 'block_analytics_summary')
+                    json.append(info)
+                return json
+
+            def publish_event(event_name, result, **kwargs):
+                """ Helper method to publish an event for analytics purposes """
+                event_data = {
+                    "location": unicode(location),
+                    "previous_count": previous_count,
+                    "result": result,
+                    "max_count": max_count
+                }
+                event_data.update(kwargs)
+                tracker.emit("edx.librarycontentblock.content.{}".format(event_name), event_data)
+
+            if block_keys['invalid']:
+                # reason "invalid" means deleted from library or a different library is now being used.
+                publish_event(
+                    "removed",
+                    result=format_block_keys(block_keys['selected']),
+                    removed=format_block_keys(block_keys['invalid']),
+                    reason="invalid"
+                )
+
+            if block_keys['overlimit']:
+                publish_event(
+                    "removed",
+                    result=format_block_keys(block_keys['selected']),
+                    removed=format_block_keys(block_keys['overlimit']),
+                    reason="overlimit"
+                )
+
+            if block_keys['added']:
+                publish_event(
+                    "assigned",
+                    result=format_block_keys(block_keys['selected']),
+                    added=format_block_keys(block_keys['added'])
+                )
+
+        def check_child_removal(block_key):
+            """
+            Check if selected block should be removed.
+            Block is removed if it is part of library_content, but has not been selected
+            for current user.
+            """
+            if block_key not in children:
+                return False
+            if block_key in children and block_key in selected_children:
```

Comment:
Can replace this with:
```
if block_key in selected_children:
```
since the first phrase is always going to be true because of the previous line.